diracx-routers/src/diracx/routers/__init__.py

```diff
@@ -446,3 +452,50 @@ async def db_transaction(db: T2) -> AsyncGenerator[T2, None]:
         if reason := await is_db_unavailable(db):
             raise DBUnavailable(reason)
         yield db
+
+
+class ClientMinVersionCheckMiddleware(BaseHTTPMiddleware):
+    """Custom FastAPI middleware to verify that
+    the client has the required minimum version.
+    """
+
+    def __init__(self, app: FastAPI):
+        super().__init__(app)
+        self.min_client_version = get_min_client_version()
+
+    async def dispatch(self, request: Request, call_next) -> Response:
+        client_version = request.headers.get("DiracX-Client-Version")
+        if client_version and self.is_version_too_old(client_version):
+            # When comes from Swagger or Web, there is no client version header.
+            # This is not managed here.
+            raise HTTPException(
+                status_code=HTTPStatus.UPGRADE_REQUIRED,
+                detail=f"Client version ({client_version}) not recent enough (>= {self.min_client_version}). Upgrade.",
+            )
+
+        response = await call_next(request)
+        return response
+
+    def is_version_too_old(self, client_version: str) -> bool | None:
+        """Verify that client version is ge than min."""
+        try:
+            return parse(client_version) < parse(self.min_client_version)
+        except InvalidVersion as iv_exc:
+            logger.info(iv_exc)
```

Comment:
This will be printed on server side, the client will not have a meaningful error on its side. Is this expected?
May be you should not catch this error or catch it but raise another `HTTPException` (`Invalid client version`) in this case?
And then you can adjust the test accordingly (make sure it raises an exception instead of capturing the stdout/stderr.